src/components/MapView/Legends/index.tsx

```diff
@@ -76,35 +89,113 @@ function Legends({ classes, layers }: LegendsProps) {
 }
 
 // Children here is legendText
-function LegendItem({ classes, title, legend, children }: LegendItemProps) {
+function LegendItem({
+  classes,
+  id,
+  title,
+  legend,
+  type,
+  opacity: initialOpacity,
+  children,
+}: LegendItemProps) {
+  const map = useSelector(mapSelector);
+  const [opacity, setOpacityValue] = useState<number | number[]>(
+    initialOpacity || 0,
+  );
+
+  const [toggle, setToggle] = useState(true);
+  const toggleChecked = (prev: Boolean) => {
+    setToggle(!prev);
+  };
+
+  const handleChangeOpacity = (event: object, newValue: number | number[]) => {
+    setOpacityValue(newValue);
+  };
+
+  const layerTypes = ['wms', 'nso', 'point_data', 'impact'];
+  const [layerId, opacityType] = (e => {
+    switch (e) {
+      case 'wms':
+        return [`layer-${id}`, 'raster-opacity'];
+      case 'impact':
+      case 'nso':
+        return [`layer-${id}-fill`, 'fill-opacity'];
+      case 'point_data':
+        return [`layer-${id}-circle`, 'circle-opacity'];
+      default:
+        return ['', ''];
+    }
+  })(type);
+
+  const allLayers = map!.getStyle().layers;
```

Comment:
Instead of assuming these are not null, it would be nice to just render nothing or a message just in case they are

Reply:
i think it will not be null since there are many initial layers like boundaries, line, etc and if the map layer is not found in the list, the opacity feature should not be rendered

Reply:
got it, but as it is now, an exception might be thrown if we somehow end up with a null map. Doing something like:
```ts
if(!map){
return null;
} 
```
would be good enough :+1: